app/components/channel_drawer_list/channel_drawer_list.js

```diff
@@ -51,23 +65,40 @@ class ChannelDrawerList extends Component {
         this.firstUnreadChannel = null;
         this.lastUnreadChannel = null;
         this.state = {
+            dataSource: this.buildData(props),
+            searching: false,
             showAbove: false,
             showBelow: false,
-            dataSource: this.buildData(props)
+            term: null
         };
+
         MaterialIcon.getImageSource('close', 20, this.props.theme.sidebarHeaderTextColor).
         then((source) => {
             this.closeButton = source;
         });
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
+        const currentProps = {...this.props};
+        const props = {...nextProps};
+
+        // remove the profiles, otherChannels, myPreferences and statuses
+        // to prevent the component to re-render every time they change
+        Reflect.deleteProperty(currentProps, 'profiles');
```

Comment:
I think this is a sign that we should break up this component a bit. It seems pretty weird that this component would take in all these props, but not care if they change.

Like for the statuses, the actual row component could just get that from the store itself so that this wouldn't update whenever they change. Same with the profiles; we could just pass in a user id or channel id (for group messages), and the row component could figure out the users and names it needs to render